napari/layers/_pyramid_layer/model.py

```diff
@@ -0,0 +1,230 @@
+import numpy as np
+from copy import copy
+
+from ...util.event import Event
+
+from .._register import add_to_viewer
+from .._image_layer import Image
+from .._image_layer.view import QtImageLayer
+from .._image_layer.view import QtImageControls
+
+
+@add_to_viewer
```

Comment:
I would look into instead adding a `__new__` method in the `Image` class that returns a `Pyramid` object if `pyramid=True` (see [this blog post](https://spyhce.com/blog/understanding-new-and-init)). Don't forget to import `Pyramid` inside `__new__` so as to avoid circular references :)

Reply:
@kne42 fascinating! I was unaware of this. As such I am hesitant to follow that pattern. But not totally opposed. =)